src/analyzer/analyzerqueue.cpp

```diff
@@ -299,12 +302,21 @@ void AnalyzerQueue::run() {
 }
 
 void AnalyzerQueue::execThread() {
-    const mixxx::DbConnectionPooler dbConnection(m_pDbConnectionPool);
-    if (!dbConnection) {
-        kLogger.warning()
-                << "Failed to open database connection for analyzer queue";
-        kLogger.debug() << "Exiting thread";
-        return;
+    mixxx::DbConnectionPooler dbConnectionPooler;
+    if (m_pAnalysisDao) {
+        // Only create/open a new database connection for when needed
+        // for storing waveform analyses
```

Comment:
How about:  
// m_pAnalysisDao ramains null if no analyzer needs database access. 
// currently only waveform analyses makes use of it.